wallet/src/repository/wallet/mod.rs

```diff
@@ -1095,16 +1108,53 @@ where
             vec![],
             Some(&request),
             fee,
+            fee_type,
             state,
             timestamp,
             tx_pending_timeout,
             None,
             utxo_strategy,
+            self.params.max_dr_weight,
         )?;
 
         Ok((inputs, outputs))
     }
 
+    /// Function that returns an address for the change ValueTransferOutput
+    fn calculate_change_address(
+        &self,
+        is_vtt: bool,
+        inputs: &[Input],
+        state: &mut State,
+    ) -> Result<PublicKeyHash> {
+        let pkh = if is_vtt {
+            // In case of VTTransaction, a new internal address is used
+            self._gen_internal_address(state, None)?.pkh
+        } else {
+            // In case of DRTransaction, the first input pkh will be used
+            let first_input = inputs.first().clone().unwrap().output_pointer();
+            let key_balance = state.utxo_set.get(&first_input.into()).unwrap();
+
+            let model::Path {
+                keychain, index, ..
+            } = self.db.get(&keys::pkh(&key_balance.pkh))?;
+            let parent_key = state
+                .keychains
+                .get(keychain as usize)
+                .expect("could not get keychain");
+
+            let extended_sign_key =
+                parent_key.derive(&self.engine, &types::KeyPath::default().index(index))?;
+
+            let public_key: types::PK =
+                types::ExtendedPK::from_secret_key(&self.engine, &extended_sign_key).into();
+
+            witnet_data_structures::chain::PublicKey::from(public_key).pkh()
```

Comment:
This code is a hack from the past, do you know why do we calculate the pkh again instead of using `key_balance.pkh`?